data.md

```diff
@@ -73,11 +73,11 @@ When we are initializing a table with element segment, we need to define a list
     rule #getElemSegment(E ES, I) => #getElemSegment(ES, I -Int 1) requires I >Int 0
 ```
 
-Memories/tables can optionally have a max size which the memory may not grow beyond.
-The sort `MaxBound` provides a potentially "infinite" `Int`.
+In some cases, an integer is optional, such as when either giving or omitting the max bound when defining a table or memory.
+The sort `OptionalInt` provides this potentially "undefined" `Int`.
 
 ```k
-    syntax MaxBound ::= Int | ".MaxBound"
+    syntax OptionalInt ::= Int | ".NoInt"
```

Comment:
This should be fine, but is a little weird that `".Identifier"` means "no identifier", but `".NoInt"` means "No Int". Just wondering is `".Int"` here will be better and more consistent.

Reply:
This will drive me crazy, can we follow the naming convention please? Either `.Int` or `.OptionalInt` seem like better choices than `.NoInt`.

Reply:
Done.